ZenPacks/zenoss/LinuxMonitor/modeler/plugins/zenoss/cmd/linux/os_service.py

```diff
@@ -0,0 +1,236 @@
+    statd start/running, process 721
+    failsafe stop/waiting
+    mountall.sh start/running
+    resolvconf start/running
+    atd start/running, process 1229
+    dbus start/running, process 998
+    docker start/running, process 958
+    ...
+
+
+Systemv output('service --status-all'):
+
+    ...
+    htcacheclean is stopped
+    httpd is stopped
+    ip6tables: Firewall is not running.
+    IPsec stopped
+    iptables: Firewall is not running.
+    irqbalance (pid  1163) is running...
+    iscsi is stopped
+    iscsid is stopped
+    Kdump is not operational
+    lldpad (pid  1200) is running...
+    lvmetad is stopped
+    mdmonitor is stopped
+    messagebus (pid  1240) is running...
+    multipathd is stopped
+    mysqld is stopped
+    netconsole module not loaded
+    Configured devices:
+    lo Auto_eth0
+    Currently active devices:
+    lo eth0
+    NetworkManager (pid  1251) is running...
+    rpc.svcgssd is stopped
+    rpc.mountd is stopped
+    nfsd is stopped
+    rpc.rquotad is stopped
+    rpc.statd (pid  1270) is running...
+    nscd is stopped
+    ...
+
+
+"""
+
+
+import re
+
+from Products.DataCollector.plugins.CollectorPlugin import LinuxCommandPlugin
+
+
+__doc__ = """os_service
+Collect linux services information using appropriate init service command.
+"""
+
+
+RE_SYSTEMD_SERVICE = re.compile('(?P<title>[@A-Za-z0-9\-\.]+)\.service\s\-\s'
+                                '(?P<description>.+)\s+'
+                                'Loaded:\s(?P<loaded_status>\w.+\))\s+'
+                                'Active:\s(?P<active_status>\w+\s\(\w+\)(\ssince.+ago)?)'
+                                '(.+Main\sPID:\s(?P<main_pid>\d+))?')
+RE_UPSTART_SERVICE = re.compile('(?P<title>[A-Za-z0-9\-\.]+)\s'
+                                '(?P<active_status>[\w/]+)'
+                                '(.\sprocess\s(?P<main_pid>\d+))?')
+RE_SYSTEMV_SERVICE = re.compile('(?P<title>[A-Za-z0-9_\-\.\s:]+)'
+                                '((\s|:)\(pid\s+(?P<main_pid>\d+)\))?'
+                                '\sis\s(?P<active_status>[\w\s]+)')
+RE_PROCESS = re.compile('(?<=Process:\s)(\d+\s\w+=\S+\s([A-z\-=]+\s)*\(\S+\s\S+\))')
+
+
+def systemd_getServices(services):
+    return ''.join(map(lambda x: x or '\n', services)).splitlines()
+
+
+def systemd_getProcesses(line):
+    matches = RE_PROCESS.findall(line)
+    if matches:
+        return zip(*matches)[0]
+
+
+SERVICE_MAP = {
+    'SYSTEMD': {
+        'regex': RE_SYSTEMD_SERVICE,
+        'functions': {
+            'services': systemd_getServices,
+            'processes': systemd_getProcesses
+        }
+    },
+    'UPSTART': {
+        'regex': RE_UPSTART_SERVICE,
+    },
+    'SYSTEMV': {
+        'regex': RE_SYSTEMV_SERVICE,
+    },
+    'UNKNOWN': None
+}
+
+
+class os_service(LinuxCommandPlugin):
+
+    command = ('export PATH=$PATH:/bin:/sbin:/usr/sbin; '
+               'if command -v systemctl >/dev/null 2>&1; then '
+                'echo "SYSTEMD"; '
```

Comment:
This one and similar will be a pep8 issue, IMHO, it's much better to have some space here.